### DISCUSSION TITLE
newbie needs help, build fail

### DISCUSSION THREAD

**Discussion by moscowgentalman** [infiniflow/infinity#1161] [Q&A] [answered]
hi, there,
I have followed the instructions in the wiki page. Using a VmWare virtural machine installing Ubuntu 23.10. And step by step , copy and paste the commands in the wiki to build from source, However, I got build erros:

error: AST file was compiled with the target feature '+fma' but the current translation unit is not
error: current translation unit is compiled with the target feature '+64bit' but the AST file was not
error: current translation unit is compiled with the target feature '+aes' but the AST file was not
error: current translation unit is compiled with the target feature '+avx' but the AST file was not
error: current translation unit is compiled with the target feature '+cmov' but the AST file was not
error: current translation unit is compiled with the target feature '+crc32' but the AST file was not
error: current translation unit is compiled with the target feature '+cx16' but the AST file was not
error: current translation unit is compiled with the target feature '+cx8' but the AST file was not
error: current translation unit is compiled with the target feature '+f16c' but the AST file was not
error: current translation unit is compiled with the target feature '+fsgsbase' but the AST file was not
error: current translation unit is compiled with the target feature '+fxsr' but the AST file was not
error: current translation unit is compiled with the target feature '+mmx' but the AST file was not
error: current translation unit is compiled with the target feature '+pclmul' but the AST file was not
error: current translation unit is compiled with the target feature '+popcnt' but the AST file was not
error: current translation unit is compiled with the target feature '+rdrnd' but the AST file was not
error: current translation unit is compiled with the target feature '+sahf' but the AST file was not
error: current translation unit is compiled with the target feature '+sse' but the AST file was not
error: current translation unit is compiled with the target feature '+sse2' but the AST file was not
error: current translation unit is compiled with the target feature '+sse3' but the AST file was not
fatal error: too many errors emitted, stopping now [-ferror-limit=]

The Ubuntu info:
liuming@liuming-None:~/infinity/cmake-build-debug$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 23.10
Release:	23.10
Codename:	mantic

I don't understand the error message above, and try to clean the workspace several times, but issues are the same. Can anyone share a light on me ? I cannot access any website outside of china, so twitter or other communication channels seems not an option for me , at least now. So i tried to ask for help in this github channel.

Thank you,
Ming

**Comment by yuzhichang** [comment, marked as answer]
The CMakeLists.txt of this project requris `avx2` or `fma`. Very old laptops and VirtualBox or Vmware managed virtual machines don't support such feature. Please check `cat /proc/cpuinfo | grep avx2` and `cat /proc/cpuinfo | grep fma`.

**Comment by moscowgentalman** [comment]
Thank you! It explains, the proc info shows my VM cannot support avx2 or fma. It is a pity, I don't have the required hardware or software, My vmware is version 17, I think it should not be very old, the problems should be the desktop, which is really a cheap one not good for development. I will check with other hardware once I have chance.
